avocado/core/future/settings.py

```diff
@@ -0,0 +1,493 @@
+            else:
+                self._value = dst_type(value)
+
+
+class Settings:
+    """Settings, an experimental Avocado configuration handler.
+
+    It is a simple wrapper around configparser and argparse.
+
+    Also, one object of this class could be passed as config to plugins and
+    modules.
+
+    Please, note that most of methods and attributes here are private. Only
+    public methods and attributes should be used outside this module.
+    """
+
+    def __init__(self, config_path=None):
+        """Constructor. Tries to find the main settings files and load them.
+
+        :param config_path: Path to a config file. Useful for unittesting.
+        """
+        self._config = configparser.ConfigParser()
+        self._all_config_paths = []
+        self._config_paths = []
+        self._namespaces = {}
+
+        # 1. Prepare config paths
+        if config_path is None:
+            self._prepare_base_dirs()
+            self._append_config_paths()
+        else:
+            # Only used by unittests (the --config parses the file later)
+            self._all_config_paths.append(config_path)
+
+        # 2. Parse/read all config paths
+        self._config_paths = self._config.read(self._all_config_paths)
+
+        if not self._config_paths:
+            raise ConfigFileNotFound(self._all_config_paths)
+
+        self._merge_with_configs()
+
+    def _append_config_paths(self):
+        # Override with system config
+        self._append_system_config()
+
+        # Allow plugins to modify/extend the list of configs
+        dispatcher = SettingsDispatcher()
+        if dispatcher.extensions:
+            dispatcher.map_method('adjust_settings_paths',
+                                  self._all_config_paths)
+
+        # Override with the user's local config
+        self._append_user_config()
+
+    def _append_system_config(self):
+        self._all_config_paths.append(self._config_path_system)
+        configs = glob.glob(os.path.join(self._config_dir_system_extra,
+                                         '*.conf'))
+        for extra_file in configs:
+            self._all_config_paths.append(extra_file)
+
+    def _append_user_config(self):
+        if not os.path.exists(self._config_path_local):
+            self._create_empty_config()
+        self._all_config_paths.append(self._config_path_local)
+
+    def _create_empty_config(self):
+        try:
+            path.init_dir(self._config_dir_local)
+            with open(self._config_path_local, 'w') as config_local_fileobj:
+                content = ("# You can use this file to override "
+                           "configuration values from '%s and %s\n"
+                           % (self._config_path_system,
+                              self._config_dir_system_extra))
+                config_local_fileobj.write(content)
+        except IOError:     # Some users can't write it (docker)
+            pass
+
+    def _merge_with_configs(self):
```

Comment:
@clebergnu since that the current implementation is putting everything inside '_future' did you tried getting from there?

```python3
future = settings.get('_future')
value = future.get('foo.bar.int_value')
```
I will take a closer look today.

Reply:
@clebergnu, the idea was: Ignore configuration options that are not registered yet.

Using this logic will forces everyone to run `register_option()`. If I trust only on configuration files, I don't have all the information needed to create an "internal option'.

But, I agree with you, now, we have to call `_merge_with_configs()` (that I should convert into a public method), and ideally, we should hide this from the user.

But I tried to think quickly about this, and it looks like we have to change more things due to the nature of the import orders. Currently, `settings` is being initialized when `app.py` is called.